api/v1/pathwaysjob_types.go

```diff
@@ -0,0 +1,218 @@
+	ControllerStatus *PathwaysComponentStatus `json:"controllerStatus,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=colocate;default
+type DeploymentMode string
+
+const (
+	Colocate DeploymentMode = "colocate"
+	Default  DeploymentMode = "default"
+)
+
+// +kubebuilder:validation:Enum=tpu-v4-podslice;tpu-v5p-slice;tpu-v5-lite-podslice;tpu-v6e-slice
+type WorkerType string
+
+const (
+	tpu_v4_podslice      WorkerType = "tpu-v4-podslice"
+	tpu_v5p_slice        WorkerType = "tpu-v5p-slice"
+	tpu_v5_lite_podslice WorkerType = "tpu-v5-lite-podslice"
+	tpu_v6e_slice        WorkerType = "tpu-v6e-slice"
+)
+
+// The WorkerSpec struct lists the specifications for the
+// Pathways workers.
+type WorkerSpec struct {
+	// This will translate to a nodeSelector of the form
+	// cloud.google.com/gke-tpu-accelerator: tpu-v5-lite-podslice
+	Type WorkerType `json:"type"`
+
+	// This will translate to a nodeSelector of the form
+	// cloud.google.com/gke-tpu-topology:2x2
+	Topology string `json:"topology"`
+
+	// Number of TPU slices requested for the Pathways workers.
+	NumSlices int32 `json:"numSlices"`
+}
+
+// The ControllerSpec struct lists the specifications for the
+// Pathways controller. User workload can also be provided here.
+type ControllerSpec struct {
+	// DeploymentMode defines whether the user job and the Pathways
+	// resources (RM, proxy) must be colocated on TPUs, with the Pathways
+	// workers or not. If user chooses to "colocate", then the Pathways RM
+	// and proxy run together with the user job as a single pod.
+	// Users may opt for "default" placement where scheduler places the
+	// RM pod and the proxy pod on the CPU nodepools by default. User
+	// workload will be deployed separately, as a pod.
+	DeploymentMode DeploymentMode `json:"deploymentMode,omitempty"`
+
+	// UserPodTemplate accepts a pod composed of user's workload
+	// (and other) containers.
+	// https://pkg.go.dev/k8s.io/api/core/v1#PodTemplateSpec
+	// +optional
+	UserPodTemplate *corev1.PodTemplateSpec `json:"template,omitempty" protobuf:"bytes,6,opt,name=template"`
+}
+
+type PathwaysConditionType string
+
+// These are built-in conditions for PathwaysJob.
+const (
+	// PathwaysJobPending means the PathwaysJob is constructed and/or may be
+	// deployed, but pods are yet to be scheduled on nodes.
+	PathwaysJobPending PathwaysConditionType = "Pending"
+	// PathwaysJobRunning means PathwaysJob has been scheduled and
+	// Pathways servers have started running.
+	PathwaysJobRunning PathwaysConditionType = "Running"
+	// PathwaysJobCompleted means the underlying JobSet has completed its
+	// execution.
+	PathwaysJobCompleted PathwaysConditionType = "Completed"
+	// PathwaysJobFailed means the JobSet has failed its execution.
+	// Reason for failure may be found in Condition.Message
+	PathwaysJobFailed PathwaysConditionType = "Failed"
+	// PathwaysJobSuspended means the underlying Jobset is suspended.
+	PathwaysJobSuspended PathwaysConditionType = "Suspended"
```

Comment:
If a PathwaaysJob is Queued, is that suspended?

Reply:
PathwaysJob is suspended if the underlying JobSet is suspended (which happens when the job and pods in turn are suspended). Kueue is not involved in the PathwaysJob journey.